### PR TITLE
qadaptive avg pooling benchmarking

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29270 qadaptive avg pooling benchmarking**
* #29268 qavgpool benchmarking
* #29259 Adding longer tests for relu
* #29258 Removed the observer from the benchmarks
* #29257 Adding short tests
* #29250 qpool benchmarking
* #29183 qrelu6 benchmarking
* #29182 Creating a base benchmarking class for activations.

Differential Revision: [D18342898](https://our.internmc.facebook.com/intern/diff/D18342898)

### CODE REVIEW

File: benchmarks/operator_benchmark/pt/qpool_test.py [hunk 1]
```diff
@@ -45,6 +45,51 @@
     tags=('short',)
 )
 
+# We will use the same dimensions as maxpool
+qadaptive_long_configs = op_bench.cross_product_configs(
+    N=range(1, 16),
+    input_dim=(
+        (  1,   3,   3),  # dummy        # noqa
+        (  3,  64,  64),  # dummy        # noqa
+        (  3,  64,  64),  # dummy        # noqa
```
Comment (mingzhe09088) [inline benchmarks/operator_benchmark/pt/qpool_test.py]: duplicate shapes?

File: benchmarks/operator_benchmark/pt/qpool_test.py [hunk 2]
```diff
@@ -45,6 +45,51 @@
...
+        ( 64, 224, 224),  # MaxPool2d-4  # noqa
+        (128, 112, 112),  # MaxPool2d-9  # noqa
+        (256,  56,  56),  # MaxPool2d-16 # noqa
+        (512,  28,  28),  # MaxPool2d-23 # noqa
+        (512,  14,  14),  # MaxPool2d-30 # noqa
+    ),
+    output_dim=(
+        (  1,   3,   3),  # dummy        # noqa
+        (  3,  64,  64),  # dummy        # noqa
+        (  3,  64,  64),  # dummy        # noqa
```
Comment (mingzhe09088) [inline benchmarks/operator_benchmark/pt/qpool_test.py]: ditto

File: benchmarks/operator_benchmark/pt/qpool_test.py [hunk 3]
```diff
@@ -89,10 +134,22 @@ def init(self, N, C, H, W, k, s, p, ceil, contig, dtype):
...
+    def init(self, N, input_dim, output_dim, dtype, contig):
+        input_dim = (N,) + input_dim
+        output_dim = (N,) + output_dim
+        self.pool_op = torch.nn.AdaptiveAvgPool2d(output_dim)
+        super(QAdaptiveAvgPool2dBenchmark, self).setup(*input_dim, dtype,
+                                                       contig)
+
+
+op_bench.generate_pt_test(qadaptive_short_configs, QAdaptiveAvgPool2dBenchmark)
+op_bench.generate_pt_test(qpool2d_short_configs, QAvgPool2dBenchmark)
```
Comment (mingzhe09088) [inline benchmarks/operator_benchmark/pt/qpool_test.py]: Concatenate configs. See comments on other PRs.

**Comment by z-a-f** [comment]
Sorry -- this was a dup submission. My bad!

**Pull request closed without merging**